File: day1/day1.py
# part 1

with open("input") as f:
    current_calories = calories_high = 0
    while (line := f.readline()):
        if len(line) == 1:
            if calories_high < current_calories:
                calories_high = current_calories
            current_calories = 0
        else:
            current_calories = current_calories + int(line)
    print(calories_high)

File: day1/day1.rb
# part 1

current = 0
highest = 0
File.foreach 'input' do |line| 
  current = current + line.to_i
  if line == "\n"
    if highest < current
      highest = current 
	end
  	current = 0
  end
end
p highest

# part 2

current = 0
highest = []
File.foreach 'input' do |line| 
  current = current + line.to_i
  if line == "\n"
    highest.append(current)
  	current = 0
  end
end
highest.sort!
sum = highest[-1] + highest[-2] + highest[-3]
p sum
